pages/register.tsx

```diff
@@ -0,0 +1,395 @@
+                  pattern="(?:\+[0-9]{2,3}-[0-9]{1,15})|(?:[0-9]{8})"
+                />
+              </Control>
+            </Field>
+            <Field>
+              <Label>College</Label>
+              <Control>
+                <div>
+                  <ReactSelect
+                    value={getCollege()}
+                    options={mapCode(colleges)}
+                    onChange={(input: { value: string }): void => {
+                      setCollegeCode(input.value);
+                    }}
+                  />
+                  <input
+                    tabIndex={-1}
+                    autoComplete="off"
+                    style={{ position: "absolute", opacity: 0, height: 0 }}
+                    value={collegeCode}
+                    onChange={() => {}}
+                    required
+                  />
+                </div>
+              </Control>
+            </Field>
+            <Field>
+              <Label>Major</Label>
+              <div>
+                <ReactSelect
+                  value={getMajor()}
+                  options={mapCode(majors)}
+                  onChange={(input: { value: string }): void => {
+                    setMajorCode(input.value);
+                  }}
+                />
+                <input
+                  tabIndex={-1}
+                  autoComplete="off"
+                  style={{ position: "absolute", opacity: 0, height: 0 }}
+                  value={majorCode}
+                  onChange={() => {}}
+                  required
+                />
+              </div>
+            </Field>
+            <Field>
+              <Label>Year of Entry</Label>
+              <Control>
+                <div>
+                  <ReactSelect
+                    value={getTermStart([
+                      -8,
+                      -7,
+                      -6,
+                      -5,
+                      -4,
+                      -3,
+                      -2,
+                      -1,
+                      0,
+                    ]).find((term) => term.value === doEntry)}
+                    options={getTermStart([-8, -7, -6, -5, -4, -3, -2, -1, 0])}
+                    onChange={(input: { value: string }): void => {
+                      setDoEntry(input.value);
+                    }}
+                  />
+                  <input
+                    tabIndex={-1}
+                    autoComplete="off"
+                    style={{ position: "absolute", opacity: 0, height: 0 }}
+                    value={doEntry}
+                    onChange={() => {}}
+                    required
+                  />
+                </div>
+              </Control>
+            </Field>
+            <Field>
+              <Label>Expected Graduation Year</Label>
+              <Control>
+                <div>
+                  <ReactSelect
+                    value={getTermEnd([-2, -1, 0, 1, 2, 3, 4, 5, 6, 7, 8]).find(
```

Comment:
use `useMemo` and store the value of getTermEnd(...) and reuse the stored value, not calculate it every time